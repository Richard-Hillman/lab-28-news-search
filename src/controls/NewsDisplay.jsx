/* eslint-disable max-len */
import React from 'react';
import NewsSearch from '../presentational/NewsSearch';
import NewsList from '../presentational/NewsList';
import { getArticles } from '../services/Services';

export default class NewsDisplay extends React.Component {
    
    state= {
      searchText: '',
      searchResults: []
    }

    componentDidMount() {
      getArticles()
        .then(searchResults => this.setState({ searchResults }));
    }

    handleClick = (e) => {
      this.setState({ searchText: e.target.value });
    }  

    handleSubmit = async(e) => {
      e.preventDefault();
      await getArticles(this.state.searchText)
        .then(searchResults => this.setState({ searchResults }));
    } 

    render() {
      return (
      // eslint-disable-next-line max-len
        <div style={{ backgroundColor: 'black', width: '90vh', height: '90vw', color: 'white' }}>
    
          <h1> Welcome to the NEWS</h1>

          <NewsSearch  
            searchText = {this.state.searchText } 
            handleClick = { this.handleClick } 
            handleSubmit = { this.handleSubmit } 
          /> 
  
          <NewsList searchResults = { this.state.searchResults } />

        </div>
      );
    }
}


