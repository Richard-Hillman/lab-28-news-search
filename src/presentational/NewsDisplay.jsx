import React from 'react';
import PropTypes from 'prop-types';
import NewsSearch from './NewsSearch';

const NewsDisplay = () => (
  <div style={{ backgroundColor: 'black', width: '90vh', height: '90vw', color: 'white' }}>
    
    <h1> Welcome to the NEWS</h1>

    <NewsSearch /> 
    
    <input default="search the news" style={{ fontColor: '' }}></input>
  
  </div>
);

NewsDisplay.PropTypes = {
  search: PropTypes.string.isRequired 
};

export default NewsDisplay;
