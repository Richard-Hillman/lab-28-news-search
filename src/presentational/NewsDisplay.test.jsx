import React from 'react';
import { render, cleanup } from '@testing-library/react';
import NewsDisplay from '../controls/NewsDisplay';

describe('NewsDisplay component', () => {
  afterEach(() => cleanup());
  it('renders NewsDisplay', () => {
    const { asFragment } = render(<NewsDisplay />);
    expect(asFragment()).toMatchSnapshot();
  });
});
