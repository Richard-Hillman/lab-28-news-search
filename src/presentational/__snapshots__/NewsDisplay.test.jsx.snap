// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsDisplay component renders NewsDisplay 1`] = `
<DocumentFragment>
  <div
    style="background-color: black; width: 90vh; height: 90vw; color: white;"
  >
    <h1>
       Welcome to the NEWS
    </h1>
    <div
      style="background-color: grey; width: 80vh; height: 20vw; color: black;"
    >
      <h1
        style="color: yellow;"
      >
         Search the NEWS
      </h1>
      <input
        default=""
      />
      <button>
         Search 
      </button>
    </div>
    <div
      style="background-color: black; width: 50vh; height: 70vw; color: white;"
    >
      <h1>
         Results 
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
